### PR TITLE
Uladzislau_Pinchuk_Task_1

### PR DESCRIPTION
**Pull request by Killerkip10**

### CODE REVIEW

File: dev-server/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const express = require('express');
+const path = require('path');
+
+const app = express();
+
+app.use(express.static(path.join(__dirname, '../dist')));
```
Comment (NikitaSedyx) [inline dev-server/index.js]: there is simple node static server for this purpose

**Comment by NikitaSedyx** [comment]
could you please make your app available via github pages?

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const { Router } = require('./router');
+const { ArticleList, ArticleInfo } = require('./components');
+
+const router = new Router('app')
+  .add('/articles/:topic', ArticleList)
+  .add('/article/:topic/:id', ArticleInfo)
+  .init();
+
+button1.addEventListener('click', () => router.push('/articles/books'));
```
Comment (NikitaSedyx) [inline src/index.js]: o_O? I need to click a button to open the list?

File: src/router/router.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    this.componentDidMount();
+  };
+
+  componentRender = () => {
+    this.componentWillRender();
+    this.#rootElement.innerHTML = this.#component.render();
+    this.componentWasRender();
+  };
+
+  componentWillRender = () => {
```
Comment (NikitaSedyx) [inline src/router/router.js]: IMO router shouldn't know anything about these methods

File: src/router/router.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    if (!route) {
+      return;
+    }
+
+    if (this.#component) {
+      this.componentWillUnmount();
+    }
+
+    this.#params = this._getParamsByPath(route.pathRegExp, path);
+    this.#component = new route.component({ apply: this.componentRender, router: this });
```
Comment (NikitaSedyx) [inline src/router/router.js]: `apply: this.componentRender` I hope/believe it will be removed after changes in router

**Comment by NikitaSedyx** [comment]
components can be break to smaller + can be separated to containers and presentation

**Comment by Killerkip10** [comment]
> could you please make your app available via github pages?

Done

File: src/router/router.js (hunk 4 above)
Comment (Killerkip10) [reply src/router/router.js]: done

File: src/router/router.js (hunk 3 above)
Comment (Killerkip10) [reply src/router/router.js]: done

File: src/index.js (hunk 2 above)
Comment (Killerkip10) [reply src/index.js]: > components can be break to smaller + can be separated to containers and presentation

done

Comment (Killerkip10) [reply src/index.js]: done

**Comment by Killerkip10** [comment]
> components can be break to smaller + can be separated to containers and presentation

done

File: dev-server/index.js (hunk 1 above)
Comment (Killerkip10) [reply dev-server/index.js]: done

File: src/components/ArticleList.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    }
+    if (this.#containerCardElement) {
+      this.#selectElement.removeEventListener('click', this.handleDetailsButtonClick);
+    }
+  };
+
+  componentWasRender = () => {
+    const [topic] = this.#router.params;
+
+    [this.#selectElement] = document.getElementsByClassName('article-list__select');
```
Comment (NikitaSedyx) [inline src/components/ArticleList.js]: seems it should be in select component - you call it view

File: src/components/component.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
+const component = Component => class {
```
Comment (NikitaSedyx) [inline src/components/component.js]: do you how it's pattern called?

Comment (Killerkip10) [reply src/components/component.js]: decorator?)

**Pull request merged**